src/Multivariate/ProductFun.jl

```diff
@@ -219,6 +238,33 @@ evaluate(f::ProductFun{S,V,SS,T},x::Number,y::Number) where {S<:UnivariateSpace,
     evaluate(f,x,:)(y)
 
 
+# TensorSpace evaluation
+function evaluate(f::TensorIteratorFun{S, d, SS, T},x...) where {S<:UnivariateSpace, d, SS, T}
+    highest_order = f.orders.n[1]
+    n = length(f.coefficients)
+
+    # this could be lazy evaluated for the sparse case
+    A = [Fun(f.space.spaces[i], [zeros(k);1])(x[i]) for k=0:highest_order, i=1:d]
+    result::T = 0
+    coef_counter::Int = 1
+    for i in f.iterator
+        tmp = f.coefficients[coef_counter]
+        if tmp != 0
+            tmp_res = 1
+            @inbounds @simd for k=1:d
```

Comment:
Does `@simd` help here, since you're indexing into another iterator? Or is the `simd` applied to the inner indexing? If the performance gain isn't significant, perhaps it'll be better to leave out `@inbounds`

Reply:
I checked it, there exists a small performance difference, but it is small. Most time is spent in the iterator, I will delete it for now.